apps/solution/src/components/EditPerson/EditPersonForm.tsx

```diff
@@ -0,0 +1,176 @@
+  const managers = useManagers(person.id);
+
+  const {
+    firstname,
+    lastname,
+    manager,
+    position,
+    email,
+    phone,
+    photo,
+    entryDate,
+    birthDate,
+    gender,
+    managerId,
+  } = formData;
+
+  const selectedManagerName =
+    managers?.find((m) => m.id === formData.managerId)?.firstname || manager;
+
+  const handleChange = (
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    const { name, value } = e.target;
+    console.log(value);
+    setFormData((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    updatePerson({
+      ...formData,
+      entryDate: unNormalizeDate(entryDate),
+      birthDate: unNormalizeDate(birthDate),
+      manager: selectedManagerName,
+    }).then(() => refreshPerson());
+  };
+
+  return (
+    <form className={styles.container} onSubmit={handleSubmit}>
+      <img
+        alt={`${firstname} ${lastname}`}
+        src={photo}
+        className={styles.photo}
+      />
+      <div className={styles.informations}>
+        <Iconized icon={solid('id-card')}>Firstname: </Iconized>
+        <input
+          type="text"
+          name="firstname"
+          value={firstname}
+          onChange={handleChange}
+        />
+
+        <Iconized icon={solid('id-card')}>Name: </Iconized>
+        <input
+          type="text"
+          name="lastname"
+          value={lastname}
+          onChange={handleChange}
+        />
+
+        <Iconized icon={solid('briefcase')}>Position: </Iconized>
+        <input
+          type="text"
+          name="position"
+          value={position}
+          onChange={handleChange}
+        />
+
+        <Iconized icon={solid('envelope')}>email: </Iconized>
+        <input
+          type="email"
+          name="email"
+          value={email}
+          onChange={handleChange}
+        />
+
+        <Iconized icon={solid('image')}>Photo: </Iconized>
+        <input type="text" name="photo" value={photo} onChange={handleChange} />
+
+        <Iconized icon={solid('person')}>Manger: </Iconized>
+        <select
+          name="managerId"
+          value={managerId}
+          onChange={(e) => handleChange(e)}
+          data-cy="select-manager"
+        >
+          {managers?.map((manager) => (
+            <option key={manager.id} value={manager.id}>
+              {manager.firstname}
+            </option>
+          ))}
+        </select>
+
+        <Iconized icon={solid('phone')}>Phone:</Iconized>
+        <input
+          type="number"
+          name="phone"
+          value={phone}
+          onChange={handleChange}
+        />
+
+        <Iconized icon={solid('birthday-cake')}>Birth date: </Iconized>
+        <input
+          type="date"
+          name="birthDate"
+          value={normalizeDate(birthDate)}
+          onChange={handleChange}
+        />
```

Comment:
I thought about it but preferred having something visually "acceptable" first. Should I wait for your component or make the changes right away? WDYT @JonathanGeoffroy  & @VictorJozwicki ?

Reply:
Well that would depend on my willpower to get back into the project when I have some time ahah. But I guess this can wait for now, I just need to make sure it's going to be done in the kanban